src/sync/impl/sync_metadata.cpp

```diff
@@ -227,12 +260,24 @@ util::Optional<SyncUserMetadata> SyncMetadataManager::get_or_make_user_metadata(
         // Check the results again.
         row = results.first();
         if (!row) {
+            // Because "making this user" is our last action, set this new user as the current user
+            TableRef currentUserIdentityTable = ObjectStore::table_for_object_type(realm->read_group(),
+                                                                                   c_sync_current_user_identity);
+
+            Row currentUserIdentityRow;
```

Comment:
Since this should not be based on `master` but `core-6` or `core-6.1`, `Row` must be replaced with `Obj`.